### PR TITLE
[Core] Stop Ray stop from killing redis that's not started by Ray

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Currently, `ray stop` logic is vulnerable, and it kills Redis server that's not started by Ray. This PR fixes the issue by better checking the executable name of redis-server (If it is redis-server created by Ray, it should contain Ray specific path copied while wheels are built).

I originally tried to obtain ppid and kill a redis-server only when it is created from the same parent, but it turns out all processes started by ray start has no ppid. 

While the best solution is to have some "process manager" that we can detect redis server started by us, I think there's no need to put lots of efforts here right now since Redis will be removed soon. We will eventually move to a better direction (process manager) to handle this sort of issues. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/scripts/scripts.py [hunk 1]
```diff
@@ -827,6 +827,13 @@ def stop(force):
             corpus = (proc_cmd
                       if filter_by_cmd else subprocess.list2cmdline(proc_args))
             if keyword in corpus:
+                # This is a way to avoid killing redis server that's not started by Ray.
+                # We are using a simple hacky solution here since
+                # Redis server will anyway removed soon from the ray repository.
+                if (keyword == "redis-server"
+                        and "core/src/ray/thirdparty/redis/src/redis-server"
```
Comment (scv119) [inline python/ray/scripts/scripts.py:834]: this looks a bit fragile too... i.e. it probably won't work with windows, and subject to failure if the path changes.

Comment (scv119) [inline python/ray/scripts/scripts.py:834]: this looks a bit fragile too... i.e. it probably won't work with windows, and subject to failure if the path changes.

Comment (rkooo567) [reply python/ray/scripts/scripts.py:834]: I think we don't need to worry about path changes. Redis will be removed from Ray in a couple months, and this path hasn't changed for a very long time (more than 2 years, this was the same). It is also hard-coded here https://github.com/ray-project/ray/blob/a60251f47a6f675a8ae4fb8ef33534a79f6ac92c/python/setup.py#L139

For Windows, I will add tests and see if it works on Windows

File: python/ray/tests/test_multi_node_3.py [hunk 2]
```diff
@@ -482,6 +482,34 @@ def remote_print(s, file=None):
     assert driver2_out_split[1][-1] == "4", driver2_out_split
 
 
+@pytest.fixture
+def redis_proc():
+    """Download external redis and start the subprocess."""
+    wd = os.getcwd()
+    check_call_subprocess(
+        ["wget", "https://download.redis.io/releases/redis-6.2.6.tar.gz"])
```
Comment (scv119) [inline python/ray/tests/test_multi_node_3.py:490]: maybe just copy the existing redis-server to a tmp folder and start from there?

**Review by scv119 — APPROVED** [review]
please address the comment in test; otherwise it will become flaky

**Comment by rkooo567** [comment]
Good point about flakiness (better not having external network-relied tests). I will modify the test

**Comment by rkooo567** [comment]
On it!

**Pull request merged**